controllers/component_build_controller.go

```diff
@@ -286,67 +287,123 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 
 	// Check if Pipelines as Code workflow enabled
-	if val, exists := component.Annotations[PaCProvisionAnnotationName]; exists {
-		if val != PaCProvisionRequestedAnnotationValue && !isSwitchedImageRegistry {
-			if !(val == PaCProvisionDoneAnnotationValue || val == PaCProvisionErrorAnnotationValue) {
-				message := fmt.Sprintf(
-					"Unexpected value \"%s\" for \"%s\" annotation. Use \"%s\" value to do Pipeline as Code provision for the Component",
-					val, PaCProvisionAnnotationName, PaCProvisionRequestedAnnotationValue)
-				log.Info(message)
-			}
-			// Nothing to do
-			return ctrl.Result{}, nil
+	if pacAnnotationValue, exists := component.Annotations[PaCProvisionAnnotationName]; exists {
```

Comment:
I was thinking on that, but since both PaC and simple workflow share many local variables as context, we have to pass many arguments in both functions, which is the main reason for me to not to move the logic into separate functions now.